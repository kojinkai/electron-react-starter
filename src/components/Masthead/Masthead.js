import React, { Component } from 'react';
import './Masthead.css';
import logo from './logo.svg';

class Masthead extends Component {
  render() {
    return (
      <header className="masthead__header">
        <img src={logo} className="masthead__logo" alt="logo" />
        <h1 className="masthead__title">Welcome to Electron React</h1>
      </header>
    )
  }
}

export default Masthead;
