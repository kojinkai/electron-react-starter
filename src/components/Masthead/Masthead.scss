.masthead {
  text-align: center;
}

.masthead__logo {
  animation: masthead-logo-spin infinite 2s linear;
  height: 80px;
}

.masthead__header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.masthead__title {
  font-size: 1.5em;
}

@keyframes masthead-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
