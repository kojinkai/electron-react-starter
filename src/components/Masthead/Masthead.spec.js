import React from 'react';
import { shallow } from 'enzyme';
import Masthead from './Masthead';

it('renders according to the snapshot', () => {
  const wrapper = shallow(<Masthead />);
  expect(wrapper).toMatchSnapshot();
});
