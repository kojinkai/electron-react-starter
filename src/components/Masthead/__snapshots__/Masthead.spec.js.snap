// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders according to the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Masthead />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="masthead__header"
>
          Electron React Starter Kit
</h1>,
        <img
          alt="logo"
          className="masthead__logo"
          src="logo.svg"
/>,
      ],
      "className": "masthead",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Electron React Starter Kit",
          "className": "masthead__header",
        },
        "ref": null,
        "rendered": "Electron React Starter Kit",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "logo",
          "className": "masthead__logo",
          "src": "logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="masthead__header"
>
            Electron React Starter Kit
</h1>,
          <img
            alt="logo"
            className="masthead__logo"
            src="logo.svg"
/>,
        ],
        "className": "masthead",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Electron React Starter Kit",
            "className": "masthead__header",
          },
          "ref": null,
          "rendered": "Electron React Starter Kit",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "masthead__logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
